### ISSUE TITLE
Anytype Linux client failed to resolve when using systemd-resolved

### ISSUE THREAD

**Issue by clems4ever** [anyproto/anytype-heart#260]
### Have you read a contributing guide?

- [X] I have read CONTRIBUTING.md
- [X] I have searched the existing issues and didn't find any that were similar
- [X] I have considered creating a pull request with fixes instead of a bug report and want to proceed

### Current Behavior

As a new user, I wanted to create a space on my linux pc and login with my Android phone. It turned out that the mobile was never able to join the linux pc because there were issues preventing the linux client to contact the public server and create the account publicly. Everything went smoothly during the creation of the space in the UI (no error message), however there were some weird logs.

After I clicked on "join" and get passed the "This is the Void" step, I could observe the following logs. The first one is the one I saw first and the other ones I saw when enabling more logs.

```
23:25:04.175 › {"level":"WARN","ts":"2023-08-03T23:25:04.174+0200","logger":"anytype-grpc-server","msg":"grpc unary request is taking too long","method":"/anytype.ClientCommands/AccountCreate","in_progress":true,"goroutines":"H4sIAA
```
and
```
22:58:52.794 › {"level":"INFO","ts":"2023-08-03T22:58:52.793+0200","logger":"net.peerservice","msg":"can't connect to host","rootOp":"periodicCall","addr":"prod-any-sync-coordinator3.toolpad.org:443","error":"dial tcp: lookup prod-any-sync-coordinator3.toolpad.org: Try again"}
{"level":"INFO","ts":"2023-08-03T22:58:52.793+0200","logger":"net.peerservice","msg":"can't connect to host","rootOp":"periodicCall","addr":"prod-any-sync-coordinator3.toolpad.org:443","error":"dial tcp: lookup prod-any-sync-coordinator3.toolpad.org: Try again"}
22:58:52.795 › {"level":"INFO","ts":"2023-08-03T22:58:52.794+0200","logger":"net.peerservice","msg":"can't connect to host","addr":"prod-any-sync-filenode3.toolpad.org:443","error":"dial tcp: lookup prod-any-sync-filenode3.toolpad.org: Try again"}
{"level":"INFO","ts":"2023-08-03T22:58:52.794+0200","logger":"net.peerservice","msg":"can't connect to host","addr":"prod-any-sync-filenode3.toolpad.org:443","error":"dial tcp: lookup prod-any-sync-filenode3.toolpad.org: Try again"}
```

It turns out that after some hours of investigation by dichotomy with another PC where the client worked, I found the problem and fixed the issue by editing /etc/resolv.conf which was empty and added the cloudflare nameserver (1.1.1.1) manually. Then after a restart of the app, it started to work properly. My DNS was working on the rest of my pc before that because I had an internet connection, was able to browse websites and I was able to ping public servers.
For information I'm running Archlinux with my DNS being systemd-resolved and when I installed the box, I kept the default /etc/resolv.conf which is empty and that was ok until I tried Anytype. I faced the same issue after I tried to create a symbolic link the stub of systemd-resolved as suggested in https://wiki.archlinux.org/title/systemd-resolved.

I had a look at the code and it seems you are using a custom resolver that might cause the troubles, maybe it does not support systemd-resolved properly. What is the reason for using a custom resolver?

I'm open to discuss any potential idea/design of solution you might have and potentially give a hand on the implementation. Just let me know.

### Expected Behavior

No issue with DNS resolution when using the default DNS setup of systemd-resolved in Archlinux, i.e., I would ideally not manually mess up with /etc/resolv.conf to fix the issue, it should rather work by default when using system-resolved resolver.

### Steps To Reproduce

- Download the linux version 0.33.3 of anytype from the website
- Make sure your linux box is using systemd-resolved
- Make sure the /etc/resolv.conf is empty or points to the systemd-resolved stub.
- Make sure the DNS works on your linux box by pinging any public address like google.com.
- Run Anytype client, click "join" to create a space, get passed the "This is the Void" step until you reach the encryption key.
- Observe the logs, you should see 
22:58:52.794 › {"level":"INFO","ts":"2023-08-03T22:58:52.793+0200","logger":"net.peerservice","msg":"can't connect to host","rootOp":"periodicCall","addr":"prod-any-sync-coordinator3.toolpad.org:443","error":"dial tcp: lookup prod-any-sync-coordinator3.toolpad.org: Try again"}

The other errors are appearing when enabling more verbose logs: 
```
ANYTYPE_LOG_LEVEL="grpc=DEBUG" ANYTYPE_GRPC_LOG=2 ./Anytype-0.33.3.AppImage
```

### Environment

```markdown
- OS: Archlinux, 6.4.7-arch1-3
- Version: Anytype Linux client 0.33.3 freshly downloaded from the website.
```

**Comment by deff7** [comment]
Hello! Can you try to force native DNS resolver?

```
GODEBUG=netdns=cgo ANYTYPE_LOG_LEVEL="grpc=DEBUG" ANYTYPE_GRPC_LOG=2 ./Anytype-0.33.3.AppImage
```

https://pkg.go.dev/net#hdr-Name_Resolution

**Comment by clems4ever** [comment]
Hello, sorry I forgot to mention that I also tried that and it does not work either. That's how I came up to the conclusion that there is some custom code for the resolver and I ended up looking into the code.

**Comment by clems4ever** [comment]
Ok so apparently this works though
```
GODEBUG=netdns=go ANYTYPE_LOG_LEVEL="grpc=DEBUG" ANYTYPE_GRPC_LOG=2 ./Anytype-0.33.3.AppImage
```

Notice the netdns=go vs netdns=cgo

**Comment by deff7** [comment]
Did you try to resolve IP for host from error manually? Like `dig prod-any-sync-coordinator3.toolpad.org`

**Comment by clems4ever** [comment]
Yes, I did and it worked when I did it. As I explained, my DNS resolver before the change was fully working for all other apps on my pc. AnyType is the only one for which I have to do this workaround for it to work properly.

**Comment by cheggaaa** [comment]
We don't use any custom resolvers...
Can you run this tool https://github.com/anyproto/any-sync-tools/tree/main/any-sync-netcheck ?

**Comment by clems4ever** [comment]
When I looked I found that piece of code: https://github.com/anyproto/anytype-heart/blob/bfdcf05538c7bab1a1b407639a451df1fe745b7c/pkg/lib/net/resolver/resolver.go#L46. I have not deeply tested if it is involved but it seems a good candidate to me.

I will run the tool as soon as possible.

**Comment by clems4ever** [comment]
You are probably right because I pinpointed the code that raises the errors and they are unrelated to this Resolver, and I think it definitely comes from net.Dial somehow.

- https://github.com/anyproto/any-sync/blob/80033d3c07815f47c589b710cc7b98b339b76019/net/pool/pool.go#L83
- https://github.com/anyproto/anytype-heart/blob/e7e246db74d1dda47691d89530f3600238e8260f/core/filestorage/rpcstore/clientmgr.go#L162
- https://github.com/anyproto/any-sync/blob/80033d3c07815f47c589b710cc7b98b339b76019/net/peerservice/peerservice.go#L71C35-L71C35

I tried netcheck, and it actually works when I empty /etc/resolv.conf. After further testing, it seems that creating the symlink from the resolved stub actually works. I don't know why it did not work the first time but anyway it seems to work now. I still don't know exactly why but I will stop investigating further.

Sorry for that, at least this will be instructive for those facing the same issue, they now know that an empty /etc/resolv.conf might break the app. They need to use the resolved stub instead.
